apis/Google.Cloud.Spanner.Data/docs/index.md

```diff
@@ -32,6 +32,35 @@ If you need to provide other credentials, there are three alternatives:
 - Pass a `ChannelCredentials` into the constructor of `SpannerConnection`
   or `SpannerConnectionStringBuilder`.
 
+## SpannerConnection and Dependency Injection
+
+A `SpannerConnection` can be provided via dependency injection based
+on the ConnectionStrings section of the application configuration,
+via the `AddSpannerConnection` extension method. For example, 
+consider an application configured with `appsettings.json` file like this:
+
+```json
+{
+  "ConnectionStrings": {
+    "SpannerDb": "Data Source=projects/my-project/instances/my-instance"
+  }
+}
+```
+
+The `SpannerConnection` can be configured on the service collection
+as shown below:
+
+{{sample:SpannerConnection.DependencyInjection}}
+
+The dependency is registered as a transient dependency (as a new
+`SpannerConnection` should be created for each request), and is
+automatically disposed by the dependency injection infrastructure.
+
+If the dependency injection container has been configured with
+`ChannelCredentials` or `GoogleCredential` values, those are used
+automatically by `AddSpannerConnection`. (If both are specified,
+`ChannelCredentials` takes priority.)
```

Comment:
Mention here also that either takes priority over what's specified on the connection string.
And also, it wouldn't be bad to mention that we wouldn't recommend configuring the credential in the connection string if the connection string is going to be used through DI, as that's the case in which we get a new pool per connection, if I understood thinks correctly.

Reply:
Right, that last part probably makes it worth mentioning. (Otherwise I'd be happy to gloss over it.)

Reply:
Actually, it's okay to specify CredentialsFile in the connection string due to the way it's used. But I'm adding a note.